pipelines/copy-number-variation-vector/gcp/HMM.wdl

```diff
@@ -0,0 +1,335 @@
+version 1.0
+
+## Copyright Wellcome Sanger Institute, Oxford University, and the Broad Institute 2021
+##
+## This WDL pipeline implements the Read-Backed component of the Mospquito Phasing Pipeline as described in
+## https://github.com/malariagen/pipelines/blob/master/docs/specs/phasing-vector.md
+##
+
+
+workflow HMM {
+  String pipeline_version = "1.0.0"
+
+  input {
+    #windowed coverage parameters
+    File input_bam
+    File input_bai
+    String sample_id
+    String output_dir
+    Int interval
+    Int window_size
+    Int min_qual
+    #coverage summary parameters
+    Float accessibility_threshold
+    Float mapq_threshold
+    File accessibility_mask_file
+    File mapq_file
+    File gc_content_file
+    String sample_group_id
+    String species
+    String runtime_zones
+  }
+
+  call WindowedCoverage {
+    input:
+      input_bam = input_bam,
+      input_bai = input_bai,
+      sample_id = sample_id,
+      output_dir = output_dir,
+      interval = interval,
+      window_size = window_size,
+      min_qual = min_qual,
+      runtime_zones = runtime_zones
+  }
+
+  call CoverageSummary {
+    input:
+      coverage_tarball = WindowedCoverage.output_gz,
+      accessibility_threshold = accessibility_threshold,
+      mapq_threshold = mapq_threshold,
+      accessibility_mask_file = accessibility_mask_file,
+      mapq_file = mapq_file,
+      gc_content_file = gc_content_file,
+      sample_group_id = sample_group_id,
+      output_dir = output_dir,
+      sample_id = sample_id,
+      runtime_zones = runtime_zones
+  }
+
+  call CoverageHMM {
+    input:
+      coverage_tarball = CoverageSummary.output_gz,
+      gc_content_file = gc_content_file,
+      coverage_gc = CoverageSummary.coverage_output,
+      coverage_variance = CoverageSummary.variance_output,
+      mapq_file = mapq_file,
+      mapq_threshold = mapq_threshold,
+      species =  species,
+      output_dir = output_dir,
+      sample_id = sample_id,
+      runtime_zones = runtime_zones
+  }
+  
+  output {
+    File output_gz = CoverageHMM.output_gz
+    File coverage_variance = CoverageSummary.variance_output
+  }
+}
+
+task WindowedCoverage {
+  input {
+    File input_bam
+    File input_bai
+    String sample_id
+    String output_dir
+    Int interval
+    Int window_size
+    Int min_qual
+
+    # runtime values
+    String docker = "us.gcr.io/broad-gotc-prod/cnv:1.0.0-1679431881"
+    String ram = "8000 MiB"
+    Int cpu = 16
+    # TODO: Make disk space dynamic based on input size
```

Comment:
same as above, do you still want to do this?

Reply:
I did resize disk dynamically based on input size, although I suspect this isn't totally necessary since this will only run on a single sample at a time